### ISSUE TITLE
new rule - require description explaining noqa comments

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#8453]
noqa comments are not very descriptive, especially since they don't support the more readable name and instead only allow you to specify the code (related: #1773)

it would be nice to have a rule that enforces that noqa comments have a comment above them describing the reason the error is being suppressed, eg:

incorrect:
```py
def foo():
    a = 1  # noqa: F841 # error: add a comment on the line above explaining why the error is being ignored
```

correct:
```py
def foo():
    # false positive, see https://github.com/astral-sh/ruff/issues/8453
    a = 1  # noqa: F841
```

eslint has a rule for this: https://eslint-community.github.io/eslint-plugin-eslint-comments/rules/require-description.html

**Comment by KotlinIsland** [comment]
Extremely valuable for long term project maintenance

**Comment by DetachHead** [comment]
Closing as a duplicate of #5182
